tests/unit/einstein/test_ceph.py

```diff
@@ -0,0 +1,141 @@
+import time
+import unittest
+import subprocess
+from ims.common import config
+
+config.load()
+
+import rados
+import rbd
+from ims.common import constants
+from ims.einstein import ceph
+from ims.einstein.operations import BMI
+from ims.exception import file_system_exceptions
+from ims.common.log import trace
+import logging
+
+_cfg = config.get()
+
+POOL = _cfg.fs.pool
+CEPH_IMG = "BMI_TEST_CEPH"
+TEST_NOT_EXIST_IMG_NAME = "BMI_TEST_IMG_NOT_FOUND"
+CEPH_IMG_SIZE = 1024
+CEPH_SNAP_IMG = "BMI_TEST_CEPH_SNAP"
+TEST_DATA = 'bmi' * 10
+OFFSET = 0
+CEPH_CHILD_IMG = "BMI_TEST_CEPH_CHILD_IMAGE"
+
+
+class TestImage(unittest.TestCase):
+    """Test image operations"""
+
+    @classmethod
+    def setUpClass(self):
+        self.fs = ceph.RBD(_cfg.fs, _cfg.iscsi.password)
+        self.fs.create_image(CEPH_IMG, CEPH_IMG_SIZE)
+
+    def test_get_image(self):
+        """Test get image"""
+
+        good_img = self.fs.get_image(CEPH_IMG)
+        self.assertEqual(good_img.name, CEPH_IMG)
+
+    def test_image_not_found_exception(self):
+        with self.assertRaises(file_system_exceptions.ImageNotFoundException):
+            bad_img = self.fs.get_image(TEST_NOT_EXIST_IMG_NAME)
+
+    def test_list_images(self):
+        """Test create image and list images"""
+
+        imgs = self.fs.list_images()
+        self.assertIn(CEPH_IMG, imgs)
+        command = "echo {0} | sudo -S rbd --keyring {1} --id {2} ls " \
+                  "{3}".format(self.fs.password, self.fs.keyring,
+                               self.fs.rid, self.fs.pool)
+        p = subprocess.Popen(command, shell=True, stderr=subprocess.STDOUT,
+                             stdout=subprocess.PIPE)
+        output, err = p.communicate()
+        rbd_imgs = output.split()
+        self.assertListEqual(imgs, rbd_imgs)
+
+    def test_write(self):
+        """Test write image"""
+
+        bytes_written = self.fs.write(CEPH_IMG, TEST_DATA, OFFSET)
+        data_read = self.fs.read(CEPH_IMG, OFFSET, bytes_written)
+        self.assertEqual(TEST_DATA, data_read)
+
+    @classmethod
+    def tearDownClass(self):
+        self.fs.remove(CEPH_IMG)
```

Comment:
I don't know if you know, but setUpClass and tearDownClass are executed only once per test case class which typically doesn't suit all test cases like for example your test_write case writes something to the image but that action is not undone. It may result in other test cases being effected.

I am guessing you assumed your test cases would executed in a particular, unfortunately that is not true. Each test case should be written independent of the other assuming they may execute in any order.

So I advise to change it back to setUp and tearDown. setUpClass and tearDownClass are typically used to allocate resources like db connections, instantiate objects, etc.